core/src/main/scala/org/apache/spark/h2o/converters/SparkDataFrameConverter.scala

```diff
@@ -52,7 +52,13 @@ private[h2o] object SparkDataFrameConverter extends Logging {
   def toH2OFrame(hc: H2OContext, dataFrame: DataFrame, frameKeyName: Option[String]): H2OFrame = {
     import ai.h2o.sparkling.ml.utils.SchemaUtils._
 
-    val flatDataFrame = flattenDataFrame(dataFrame)
+    val flatDataFrameOriginal = flattenDataFrame(dataFrame)
+    val numPartitions = flatDataFrameOriginal.rdd.partitions.length
+    val flatDataFrame = if (flatDataFrameOriginal.count() < numPartitions) {
+      flatDataFrameOriginal.coalesce(numPartitions)
```

Comment:
Please can you share details how `coalesce` can prevent sending empty partitions?

Reply:
Yup. The issue is as explained here  https://0xdata.atlassian.net/browse/PUBDEV-6904. My thought was that if we ensure coalesce to number of elements in the spark data frame ( if num. of elements is smaller then number of partitions) then the coalesce would ensure that at least 1 element is in each partition. But as the tests are still failing, it might not behave like that ( did not debug since then). Any idea why? The H2O fix would fix it for sure, this is just to understand the issue a bit better